sdk/purview/Azure.Analytics.Purview.DataMap/src/Generated/AtlasBusinessMetadataDef.cs

```diff
@@ -0,0 +1,124 @@
+        /// <param name="createTime"> The created time of the record. </param>
+        /// <param name="createdBy"> The user who created the record. </param>
+        /// <param name="dateFormatter"> The date format. </param>
+        /// <param name="description"> The description of the type definition. </param>
+        /// <param name="guid"> The GUID of the type definition. </param>
+        /// <param name="name"> The name of the type definition. </param>
+        /// <param name="options"> The options for the type definition. </param>
+        /// <param name="serviceType"> The service type. </param>
+        /// <param name="typeVersion"> The version of the type. </param>
+        /// <param name="updateTime"> The update time of the record. </param>
+        /// <param name="updatedBy"> The user who updated the record. </param>
+        /// <param name="version"> The version of the record. </param>
+        /// <param name="lastModifiedTS"> ETag for concurrency control. </param>
+        /// <param name="attributeDefs"> An array of attribute definitions. </param>
+        /// <param name="serializedAdditionalRawData"> Keeps track of any properties unknown to the library. </param>
+        internal AtlasBusinessMetadataDef(TypeCategory? category, long? createTime, string createdBy, AtlasDateFormat dateFormatter, string description, string guid, string name, IDictionary<string, string> options, string serviceType, string typeVersion, long? updateTime, string updatedBy, long? version, string lastModifiedTS, IList<AtlasAttributeDef> attributeDefs, IDictionary<string, BinaryData> serializedAdditionalRawData)
+        {
+            Category = category;
+            CreateTime = createTime;
+            CreatedBy = createdBy;
+            DateFormatter = dateFormatter;
+            Description = description;
+            Guid = guid;
+            Name = name;
+            Options = options;
+            ServiceType = serviceType;
+            TypeVersion = typeVersion;
+            UpdateTime = updateTime;
+            UpdatedBy = updatedBy;
+            Version = version;
+            LastModifiedTS = lastModifiedTS;
+            AttributeDefs = attributeDefs;
+            _serializedAdditionalRawData = serializedAdditionalRawData;
+        }
+
+        /// <summary> The enum of type category. </summary>
+        public TypeCategory? Category { get; set; }
+        /// <summary> The created time of the record. </summary>
+        public long? CreateTime { get; set; }
+        /// <summary> The user who created the record. </summary>
+        public string CreatedBy { get; set; }
+        /// <summary> The date format. </summary>
+        public AtlasDateFormat DateFormatter { get; set; }
+        /// <summary> The description of the type definition. </summary>
+        public string Description { get; set; }
+        /// <summary> The GUID of the type definition. </summary>
+        public string Guid { get; set; }
```

Comment:
What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0).

Reply:
This is also discussed in the typeSpec PR.
Because in the entity create scenario, we allow the customer to provide the guid as "-1" for new asset creation. This is also the Atlas behavior. So we do not model them as GUID.